### PR TITLE
Add strain exercise

### LINKED ISSUE #399: Add strain exercise
I would like to add the [strain](https://github.com/exercism/problem-specifications/tree/main/exercises/strain) exercise.

### PR DESCRIPTION
**Pull request by keiravillekode**
resolves #399

### CODE REVIEW

File: exercises/practice/strain/.meta/example.lua [hunk 1]
```diff
@@ -0,0 +1,18 @@
+local function keep(xs, pred)
+  acc = {}
```
Comment (ryanplusplus) [inline exercises/practice/strain/.meta/example.lua:2]:
```suggestion
  local acc = {}
```

File: exercises/practice/strain/strain_spec.lua [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    return false
+  end
+
+  describe('keep', function()
+    it('keep on empty list returns empty list', function()
+      assert.same({}, keep({}, always_true))
+    end)
+
+    it('keeps everything', function()
+      assert.same({1, 3, 5}, keep({1, 3, 5}, always_true))
```
Comment (ryanplusplus) [inline exercises/practice/strain/strain_spec.lua:23]: Can you format non-empty tables with a space after `{` and before `}` as in https://github.com/exercism/lua/blob/main/exercises/practice/etl/etl_spec.lua#L24?

**Review by ryanplusplus — CHANGES_REQUESTED** [review]

File: exercises/practice/strain/.meta/example.lua (hunk 1 above)
Comment (keiravillekode) [reply exercises/practice/strain/.meta/example.lua]: done

File: exercises/practice/strain/strain_spec.lua (hunk 2 above)
Comment (keiravillekode) [reply exercises/practice/strain/strain_spec.lua]: done

**Review by ryanplusplus — APPROVED** [review]

**Pull request merged**